### ISSUE TITLE
Exclude definitions in LET blocks from outline

### ISSUE THREAD

**Issue by japgolly** [alygin/vscode-tlaplus#182]
Hi! Firstly, I absolutely love this plugin! I find the archaic toolbox IDE that comes with TLA+ unusable in so many ways so thank you so much for making this so that we can have a proper, modern, and configurable editor experience! It makes a huge difference and is deeply appreciated. Thank you. :heart: 

On to the actual issue, the outline view is populated with all definitions found in a spec. This is fantastic for top-level defs but all the defs in LET blocks appear too making it way too noisy, and hard to find what I'm looking for. Definitions inside LET blocks are private and not accessible from without the top-level definition so I don't think there's any value in keeping it in the outline view. Can we filter them out?

Alternatively if you don't want to outright remove, maybe we could have a setting to allow users to filter them out?

**Comment by japgolly** [comment]
Actually after paying a bit more attention, it seems that something funky is going on. Take this action for example:

```tla
TabRecvDraftsFromWorker ==
  LET i == SeqIndexOf(network, LAMBDA m: m.type = syncWT)
  IN
    & i != 0
    & LET msg == network[i]
          t   == msg.to
          ts  == tabs[t]
          tss == AddDraftsToTab(ts, msg.drafts, msg.edit)
      IN \E ts2 \in tss:
        & tabs' = [tabs EXCEPT ![t] = ts2]
        & RecvMsg(i)
        & UNCHANGED << browsers, remote, workers >>
```

is appearing in the outline as

![Screenshot_2020-12-11_09-58-31](https://user-images.githubusercontent.com/202935/101840036-8442c180-3b97-11eb-9463-e68c7a30af06.png)

which is weird because `i` and `msg` are missing. Looks like a bug.

**Comment by alygin** [comment]
@japgolly , thanks for the kind words! I'm glad you find the extension useful.

As for the Outline View issue, I think the better solution would be to move these private symbols one level down, under the corresponding top level symbol. This is a usual approach in other languages, and this is how this view was designed to work. So, it'll look like this:

<img width="248" alt="изображение" src="https://user-images.githubusercontent.com/2101250/103516194-2301cb80-4e81-11eb-8184-43abe7526cb0.png">

Unfortunately, the parser is not perfect at the moment (and it never will, until we switch to the native toolbox parser), and there're cases when it doesn't recognize private symbols. But I think we can improve it a bit, at least for the case you provided.

**Comment by alygin** [comment]
Improvements are available in the [v1.5.2-alpha.2 pre-release](https://github.com/alygin/vscode-tlaplus/releases/tag/v1.5.2-alpha.2).

**Comment by alygin** [comment]
Improvements published in v1.5.2
